firestore/CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/firestore/admin/v2/firestore_table_admin.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/firestore/admin/v2/common.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/firestore/admin/v2/instance.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/firestore/admin/v2/table.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/firestore/v2/firestore.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/firestore/v2/data.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/iam/v1/iam_policy.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/iam/v1/policy.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/longrunning/operations.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/rpc/status.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/rpc/error_details.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/api/annotations.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/api/auth.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/api/http.proto)
+# GRPC_GENERATE_CPP_MOCKS(GRPCPP_SOURCES GRPCPP_HDRS GRPC_MOCK_HDRS
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/firestore/admin/v2/firestore_instance_admin.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/firestore/admin/v2/firestore_table_admin.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/firestore/v2/firestore.proto
+#     ${PROJECT_THIRD_PARTY_DIR}/googleapis/google/longrunning/operations.proto)
+
+# Create a library with the generated files from the relevant protos.
+# add_library(firestore_protos ${PROTO_SOURCES} ${PROTO_HDRS} ${GRPCPP_SOURCES} ${GRPCPP_HDRS})
+# target_link_libraries(firestore_protos gRPC::grpc++ gRPC::grpc protobuf::libprotobuf)
+# target_include_directories(firestore_protos PUBLIC
+#     $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+#     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+#     $<INSTALL_INTERFACE:include>)
+# target_compile_options(firestore_protos PUBLIC ${GOOGLE_CLOUD_CPP_EXCEPTIONS_FLAG})
+# add_library(firestore::protos ALIAS firestore_protos)
+
+# Enable unit tests
+enable_testing()
+
+# Capture the compiler version and the git revision into variables, then
+# generate a config file with the values.
+# if (IS_DIRECTORY ${PROJECT_SOURCE_DIR}/.git)
+#     execute_process(COMMAND git rev-parse --short HEAD
+#             OUTPUT_VARIABLE GIT_HEAD_LOG ERROR_VARIABLE GIT_HEAD_LOG)
+# else ()
+#     set(GIT_HEAD_LOG "(unknown-git-commit)")
+# endif ()
+# string(REPLACE "\n" "" GIT_HEAD ${GIT_HEAD_LOG})
+# string(TOUPPER "${CMAKE_BUILD_TYPE}" BUILD_TYPE_UPPER)
+# configure_file(client/build_info.cc.in client/build_info.cc)
+
+# include(IncludeGMock)
+
+# add_dependencies(skip-scanbuild-targets gmock firestore_protos)
+
+# All tests get added to this target below.
+add_subdirectory(google)
+add_subdirectory(client)
```

Comment:
You need to create a CMakeLists.txt file in `client/` that compiles the library.